axiom/nr_php_packages.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef NR_PHP_PACKAGES_HDR
+#define NR_PHP_PACKAGES_HDR
+
+#include "nr_log_event.h"
+#include "util_random.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+
+typedef struct _nr_php_package_t {
+  char* package_name;
+  char* package_version;
+} nr_php_package_t;
+
+typedef struct _nr_php_packages_t {
+  nr_hashmap_t* data;
+} nr_php_packages_t;
+
+/*
+ * Purpose : Create a new php package
```

Comment:
I would add docs here to explain how to create a package with an unknown version.  

You could document that passing `version = NULL`  will do this.

An alternate way which I think makes the code easier to understand would be to have a define like `PHP_PACKAGE_VERSION_UNKNOWN` which is defined as:
```
#define PHP_PACKAGE_VERSION_UNKNOWN "PHP_PACKAGE_VERSION_UNKNOWN"
```

Then to create a package with an unknown version you would do:
```
pkg = nr_php_package_create("MyFramework", PHP_PACKAGE_VERSION_UNKNOWN);
```

Either way it is important to document this use case so one doesn't have to read the function to understand how to handle it.

Reply:
I have updated the description and decided to do it the first way for now: [90ece38](https://github.com/newrelic/newrelic-php-agent/pull/754/commits/90ece38106c8b0fa9deb82bde4606dd92cd413bf)